plugin.video.hanssettings/addon.py

```diff
@@ -0,0 +1,155 @@
+'''
+    Hans Settings
+    ~~~~~~~
+
+    An XBMC addon for watching hanssettings 
+    :license: GPLv3, see LICENSE.txt for more details.
+    
+'''
+import sys
+try:
+    # For Python 3.0 and later
+    from urllib.parse import urlencode
+    from urllib.parse import parse_qsl
+    #File "*****\script.common.plugin.cache\default.py", line 30, in run
+    #  sys.path = [settings.getAddonInfo('path').decode('utf-8') + "/lib"] + sys.path
+    #  AttributeError: 'str' object has no attribute 'decode'    
+    import storageserverdummy as StorageServer
+
+except ImportError:
+    # Fall back to Python 2's urllib2
+    from urllib import urlencode
+    from urlparse import parse_qsl
+    try:
+        import StorageServer
+    except:
+        import storageserverdummy as StorageServer
+
+import resources.lib.hanssettings
+import time
+import xbmcplugin, xbmcgui, xbmcaddon
+
+PLUGIN_NAME = 'hanssettings'
+PLUGIN_ID = 'plugin.video.hanssettings'
+
+_hanssettings = resources.lib.hanssettings.HansSettings(sys.version_info[0] == 3)
+_url = sys.argv[0]
+_handle = int(sys.argv[1])
+_cache = StorageServer.StorageServer(PLUGIN_ID, 24) # (Your plugin name, Cache time in hours)
+_addon = xbmcaddon.Addon()
+
+# In deze file heb ik alle logica van kodi zitten.
+# hier worden alle files ook gecached, want dat is een kodi addon.
+
+def get_url(**kwargs):
+    """
+    Create a URL for calling the plugin recursively from the given set of keyword arguments.
+    :param kwargs: "argument=value" pairs
+    :type kwargs: dict
+    :return: plugin call URL
+    :rtype: str
+    """
+    return '{0}?{1}'.format(_url, urlencode(kwargs))
+
+def get_categories():
+    githubfiles = _cache.cacheFunction(_hanssettings.get_dataoverzicht)
+    return _hanssettings.get_overzicht(githubfiles)
+
+def get_videos(streamfile):
+    streamsdatafile = _cache.cacheFunction(_hanssettings.get_datafromfilegithub,streamfile)
+    return _hanssettings.get_items(streamsdatafile)
+
+def list_categories():
+    xbmcplugin.setPluginCategory(_handle, 'Kanalenlijst Hans (Streams)')
```

Comment:
please move hardcoded strings to localized strings in a future update

Reply:
Done